src/kw_include.sh

```diff
@@ -0,0 +1,14 @@
+# Used for file sourcing,
+# accepts as argument the path for the file to be sourced
+function include() {
+  local filename="$1"
+  local varname=${filename%.sh}
+  varname=${varname^^}_IMPORTED
+
+  if [ -n "${!varname}" ]; then
+    return 255
+  else
+    printf -v "${varname}" '%s' "1"
```

Comment:
Wouldn't `declare -g` be more readable here? See [documentation](https://www.gnu.org/software/bash/manual/bash.pdf) at pages 53, 54. Both are bash builtins, so it shouldn't matter too much. Curiosity: in my system `declare` was is about 20% faster than `printf` for this purpose.

Reply:
I agree it would be more readable, but older versions of bash don't have the -g option, @rodrigosiqueira should we care about supporting older versions or may I change this?

Reply:
I think `declare -g` is a little bit different in this context because `varname` is a local variable (and should be). Fwiu, `-g` will affect global variables, am I wrong?

Also, you said 20% fast. How did you benchmark this? Also, in terms of ms, what is the difference?

Reply:
@rodrigosiqueira You are rigth that the `-g` option affects global variables, but that is exactly the case here. We need some way to remember which files were already included, and these global variables serve this purpose. This function should assert that if a file has been included, there is a global variable associated with that variable whose value is set to 1.